### PR TITLE
⬆️  Upgrade project to yarn 2 + dependency upgrades

### PR DESCRIPTION
**Pull request by knor-el-snor**
## Known issues
- 1 test failing in the logger package. Still figuring out as no changes should've occurred there **. Please ignore this for the time being until fixed, no merge will happen before fixing this.**
 
## Changes
- Upgrade yarn 1 to yarn 2 using [zero-installs](https://yarnpkg.com/features/zero-installs)
- Use `dist` as output folder instead of `build`.
- Upgrade dependencies (only those with least amount of breaking changes) & devDependencies
- Use root `tsconfig.json` which all packages inherit from instead of copy/pasted ones per packages
- Use default synthetic import where possible due to updated typescript configuration 

## Why is this amazing
There are a lot of benefits to these changes but one of the cool things is definitely the following:

Github Actions (**before**)
Screen Shot 2021-01-30 at 23 43 00

Github Actions (**after**)
Screen Shot 2021-01-30 at 23 43 04

## How to review
- Make sure to check the CHANGELOGs as they have been fully updated with the accurate changes
- Due to the zero-installs yarn 2 upgrade it looks like a ginormous amount of files have changed. This is 95% just files in the `.yarn/cache` folder. I suggest to filter those files out on Github:
Screen Shot 2021-01-30 at 23 15 07

## How to test
- Let CI do its thing. I'll be testing it within a personal project that uses all these packages and is fully tested to see whether any unexpected errors occur all of a sudden due to these changes. Hold off on upgrading in your own projects **for now**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,7 +27,13 @@ node_modules
...
 .npm
 
-# Yarn
+# Yarn 2
+.yarn/*
+!.yarn/cache
+!.yarn/releases
+!.yarn/plugins
+!.yarn/sdks
+!.yarn/versions
```
Comment (knor-el-snor) [inline .gitignore:36]: Choice of using yarn with zero-install

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (knor-el-snor) [inline tsconfig.json:1]: Add root tsconfig.json which is being used within all packages

File: packages/translations/package.json [hunk 3]
```diff
@@ -17,11 +17,11 @@
     "url": "https://github.com/knor-el-snor/tree-house.git"
   },
   "license": "ISC",
-  "main": "build/index.js",
-  "types": "build/index.d.ts",
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
```
Comment (knor-el-snor) [inline packages/translations/package.json:21]: Replace `build` with `dist` on all packages

File: packages/logger/src/tests/logger.ts [hunk 4]
```diff
@@ -95,17 +96,17 @@ describe('Basic logger test', () => {
       getLogger().error(message);
       expect(consoleSpy).toHaveBeenCalledWith<[string]>(
         expect.stringMatching(
-          new RegExp(`^{.*\\"serviceContext\\":{\\"service\\":\\"@tree-house/logger\\",\\"version\\":\\".*-${process.env.NODE_ENV}\\"}.*}\n$`),
+          new RegExp(
+            `^{.*\\"serviceContext\\":{\\"service\\":\\"@tree-house/logger\\",\\"version\\":\\".*-${process.env.NODE_ENV}\\"}.*}\n$`,
+          ),
```
Comment (knor-el-snor) [inline packages/logger/src/tests/logger.ts:101]: @gorjuspixels For some reason this test starts failing after either upgrading yarn or aligning TS with the root tsconfig.json. Any clue why? The error is the following:

```
 Expected: StringMatching /^{.*\"serviceContext\":{\"service\":\"@tree-house\/logger\",\"version\":\".*-test\"}.*}\n$/
@tree-house/logger:     
Received: "{\"level\":\"🔥 error\",\"serviceContext\":{\"version\":\"undefined-test\"},\"namespace\":\"test\",\"eventTime\":\"1970-01-01T00:00:00.000Z\",\"context\":{},\"message\":\"Dolore voluptas voluptatem.\"}
```

File: packages/errors/src/lib/parser.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import * as _ from 'lodash';
-import * as safeJsonStringify from 'safe-json-stringify';
+import _ from 'lodash';
+import safeJsonStringify from 'safe-json-stringify';
```
Comment (knor-el-snor) [inline packages/errors/src/lib/parser.ts:2]: Used default synthetic imports from now on

File: packages/errors/src/lib/errors.ts [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
+import httpStatus from 'http-status';
 
 import { errors } from '../config/errors.config';
 
 export class ApiError<T = any> extends Error {
   code: string;
   status: number;
   i18n?: string;
-  id?: string;
+  id: string;
```
Comment (knor-el-snor) [inline packages/errors/src/lib/errors.ts:10]: As it's autogenerated it will always exist

File: packages/logger/src/tests/logger.ts (hunk 4 above)
Comment (knor-el-snor) [reply packages/logger/src/tests/logger.ts:101]: ps: I know I updated the `debug` package but the error occurred before this so that's not the reason

**Comment by yashafromrussia** [comment]
@knor-el-snor looks like the problem is with using `process.env.npm_package_name` and `process.env.npm_package_version`.. Not sure what the workaround for this is. But the values for those environment variables are not defined

**Comment by yashafromrussia** [comment]
Looks like it's been implemented but not yet released https://github.com/yarnpkg/berry/pull/2426#issuecomment-788822200

**Comment by yashafromrussia** [comment]
The [changelog](https://github.com/yarnpkg/berry/blob/master/CHANGELOG.md) doesn't mention this change either. One of the contributors is saying [it'll be released "in the next major" version](https://github.com/yarnpkg/berry/issues/1659#issuecomment-770465913). Not sure if they mean `3.x.x` which is what the `master` branch of the repo tracks... I hope that's not what they mean 😬 It does work using the master version (`yarn set version from sources`)

**Pull request merged**